File: identicon/src/main/java/com/lelloman/identicon/util/Path2.java
package com.lelloman.identicon.util;

import android.graphics.Path;

/**
 * 	a Path with a few utility methods
 */
public class Path2 extends Path {

	public void addTriangle(int x0,int y0, int x1, int y1, int x2, int y2){
		moveTo(x0,y0);
		lineTo(x1,y1);
		lineTo(x2,y2);
		lineTo(x0,y0);
	}

	public void addRectangle(int left,int top,int right,int bottom){
		moveTo(left,top);
		lineTo(right,top);
		lineTo(right,bottom);
		lineTo(left,bottom);
		lineTo(left,top);
	}

	public void addPolygon(int x0, int y0, int x1, int y1,int x2,int y2,int x3, int y3){
		moveTo(x0,y0);
		lineTo(x1,y1);
		lineTo(x2,y2);
		lineTo(x3,y3);
		lineTo(x0,y0);
	}
}

File: identicon/src/main/java/com/lelloman/identicon/util/Path2.kt
package com.lelloman.identicon.util

import android.graphics.Path

/**
 * A [Path] with a few utility methods.
 */
class Path2 : Path() {

    fun addTriangle(x0: Int, y0: Int, x1: Int, y1: Int, x2: Int, y2: Int) {
        moveTo(x0.toFloat(), y0.toFloat())
        lineTo(x1.toFloat(), y1.toFloat())
        lineTo(x2.toFloat(), y2.toFloat())
        lineTo(x0.toFloat(), y0.toFloat())
    }

    fun addRectangle(left: Int, top: Int, right: Int, bottom: Int) {
        moveTo(left.toFloat(), top.toFloat())
        lineTo(right.toFloat(), top.toFloat())
        lineTo(right.toFloat(), bottom.toFloat())
        lineTo(left.toFloat(), bottom.toFloat())
        lineTo(left.toFloat(), top.toFloat())
    }

    fun addPolygon(x0: Int, y0: Int, x1: Int, y1: Int, x2: Int, y2: Int, x3: Int, y3: Int) {
        moveTo(x0.toFloat(), y0.toFloat())
        lineTo(x1.toFloat(), y1.toFloat())
        lineTo(x2.toFloat(), y2.toFloat())
        lineTo(x3.toFloat(), y3.toFloat())
        lineTo(x0.toFloat(), y0.toFloat())
    }
}
